<?xml version="1.0"?>
<launch>
  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find husky_gazebo)/launch/playpen.launch" />

  <node pkg="husky_control"
        name="tf_broadcast"
        type="tf_broadcast.py"
        output="screen">
  </node>

  <!-- Leader -->
  <include file="$(find husky_gazebo)/launch/spawn_husky.launch">
    <arg name="robot_namespace" value="husky_0"/>
    <arg name="x" value="-5.857378868712235"/>
    <arg name="y" value="-5.131734155596889"/>
    <arg name="yaw" value="0.0"/>
  </include>
  <node pkg="husky_control"
        name="cmd_vel_keyboard"
        type="cmd_vel_keyboard.py"
        output="screen"
        args="0"> <!-- husky_0 -->
  </node>
  <!--node name="dd_kinematics" pkg="husky_control" type="dd_kinematics.py" output="screen">
  </node-->
  <node pkg="husky_control"
        name="fake_trajectory"
        type="fake_trajectory.py"
        output="screen"
        args="0"> <!-- husky_0 -->
  </node>

  <include file="$(find husky_gazebo)/launch/spawn_husky.launch">
    <arg name="robot_namespace" value="husky_1"/>
    <arg name="x" value="-6.853181263534891"/>
    <arg name="y" value="-6.253329500533115"/>
    <arg name="yaw" value="1.5"/>
  </include>
  <node name="open_nmpc_controller_1"
        pkg="open_nmpc_controller"
        type="open_mpc_controller_node"
        respawn="true"
        clear_params="true"
        output="screen"
        args="1"> <!-- husky_1-->
    <rosparam file="$(find open_nmpc_controller)/config/open_params.yaml" />
  </node>
  <include file="$(find husky_gazebo)/launch/box_filter.launch">
    <arg name="robot_namespace" value="husky_1"/>
  </include>

  <include file="$(find husky_gazebo)/launch/spawn_husky.launch">
    <arg name="robot_namespace" value="husky_2"/>
    <arg name="x" value="-7.872674591300566"/>
    <arg name="y" value="-7.352182434220175"/>
    <arg name="yaw" value="0.0"/>
  </include>
  <node name="open_nmpc_controller_2"
        pkg="open_nmpc_controller"
        type="open_mpc_controller_node"
        respawn="true"
        clear_params="true"
        output="screen"
        args="2">
    <rosparam file="$(find open_nmpc_controller)/config/open_params.yaml" />
  </node>
  <include file="$(find husky_gazebo)/launch/box_filter.launch">
    <arg name="robot_namespace" value="husky_2"/>
  </include>
</launch>
